modules/consentManagement.js

```diff
@@ -188,22 +189,29 @@ function loadConsentData(cb) {
       done(null, true, msg, ...extraArgs);
     }
   }
-  cmpCallMap[userCMP](callbacks);
+  callMap[userCMP](callbacks);
 
   if (!isDone) {
     const onTimeout = () => {
       const continueToAuction = (data) => {
         done(data, false, 'CMP did not load, continuing auction...');
       }
+
       processCmpData(provisionalConsent, {
         onSuccess: continueToAuction,
         onError: () => continueToAuction(storeConsentData(undefined))
       })
     }
     if (consentTimeout === 0) {
       onTimeout();
+    } else if (useActionTimeout && actionTimeout) {
+      if (timer != null) {
+        clearTimeout(timer);
+      }
+
+      timer = timeout(onTimeout, actionTimeout);
```

Comment:
The use case for this - from how I  interpret https://github.com/prebid/Prebid.js/issues/9176 - is a pub that says, "I want to wait a bit to get a valid consent string from the CMP, but I don't want to wait any time for the user to provide consent". That would be expressed with something like `{ actionTimeout: 0, timeout: 10000}`, meaning "wait up to 10 seconds for the user to give consent, but no longer than 0 seconds after the CMP has loaded".

With this PR you would set a timeout of 0 without waiting for the CMP to load. I think the way to do this is to start the `actionTimeout` timer here:

https://github.com/prebid/Prebid.js/blob/1960ed811bd923a445aece06b20f06e11589bf1d/modules/consentManagement.js#L86

that's when we get the first message from the CMP, which should always contain valid consent data in `provisionalConsent` (although it may deny all consent if the user hasn't clicked the button yet).

Reply:
@EskelCz is this an accurate characterization of the setting you desire? I was suspecting you wanted to lower the timeout when there is no interaction pending. 

eg { actionTimeout: 10000, timeout: 200} so that we only have the long timeout if the action is pending

Reply:
@dgirardi @patmmccann
To clarify, most of our publishers don’t actually want to wait for any interaction (often their users don't click the banner at all), but we need to wait for the CMP to be loaded, to return some consent string (either default or read from storage). Therefore, we would probably go with something like `{ actionTimeout: 0, timeout: 200 }`.

Other publishers that are willing to wait for interaction (in case of the open banner / action pending) would use `{ actionTimeout: 10000, timeout: 200 }`, the 10s for example.

If that would be a breaking change, I think even clearer naming could be used, like gdpr.actionTimeout and gdpr.loadTimeout.

@dgirardi
I’m not sure about the internal implementation, whether the `provisionalConsent` would have the consent that was read from storage.

Reply:
Just refactored the actionTimeout logic here.  if the cmp loads (within the `timeout` delay) on an initial visit (no user consent) and the actionTimeout is present in the config, the actionTimeout will be used at this point.  if the cmp doesn't load in time, the auction will just proceed.  Something to note, I did notice with the provisional consent that the consent string initially returns as an empty string.  So even if we wait for the cmp to load first (when a user has not yet consented), the consent string will get set as undefined once the auction starts.

Reply:
@EskelCz I think it depends on the CMP, not all of them will provide a "null" string. If it's working for what you're using now it should not be affected.